### PR TITLE
Added support for SpeechT5_TTS

### PR DESCRIPTION
**Pull request by ssinghalTT**
### Problem description
Added support for SpeechT5_TTS
HF link: https://huggingface.co/microsoft/speecht5_tts

### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable)
- [ ] [Galaxy quick](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-quick.yaml) CI passes (if applicable)
- [ ] [Galaxy demo tests, for Llama](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-demo-tests.yaml) CI passes, if applicable, because of current Llama work
- [ ] (For runtime and ops writers) [T3000 unit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-unit-tests.yaml) CI passes (if applicable, since this is run on push to main)
- [ ] (For models and ops writers) [T3000 demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) CI passes (if applicable, since this is required for release)
- [ ] [cpp-unit-tests](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml) job passes
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: models/experimental/speecht5_tts/tests/test_end_to_end_pcc.py [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+        "decoder_pcc": decoder_pcc,
+        "pre_postnet_pcc": pre_postnet_pcc,
+        "post_postnet_pcc": post_postnet_pcc,
+        "stop_logits_pcc": stop_logits_pcc,
+    }
+
+
+if __name__ == "__main__":
+    results = test_end_to_end_pcc()
+    print(f"\n✅ Test completed. Results: {results}")
```
Comment (jmalone-tt) [inline models/experimental/speecht5_tts/tests/test_end_to_end_pcc.py:355]: Is it intentional that we don't actually assert anything in these tests? It seems like it would be better to see a test failure instead of just printing when results are bad unless I am missing something

**Review by jmalone-tt — COMMENTED** [review]
Looks pretty good, but a question about testing

File: models/experimental/speecht5_tts/tests/test_end_to_end_pcc.py (hunk 1 above)
Comment (ssinghalTT) [reply models/experimental/speecht5_tts/tests/test_end_to_end_pcc.py:355]: @jmalone-tt thank you for the question. Currently PCC is bad for encoder. So asserting might mask if some code change fails for the future blocks. I can have an assert with low value for encoder but that just feels pointless. Since it is still experimental, I hope it is okay. I can update to add the assert if you prefer that

Comment (jmalone-tt) [reply models/experimental/speecht5_tts/tests/test_end_to_end_pcc.py:355]: I think we should at least add asserts for the other blocks, and add a comment explaining why the encoder was left out. Written as-is, there is no guarantee that other blocks will remain functional

**Review by uaydonat — APPROVED** [review]

**Review by dpopovTT — APPROVED** [review]
